src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js

```diff
@@ -53,6 +54,7 @@ const BaseAnchorForCommentsOnly = (props) => {
     return (
         <PressableWithSecondaryInteraction
             inline
+            style={[styles.cursorDefault, StyleUtils.getFontSizeStyle(props.style.fontSize)]}
```

Comment:
@mollfpr @cristipaval This PR was reverted because of the special handling of internal links explained by @parasharrajat above.

I found a solution -
basically for internal links since they are rendered as `div` we can explicitly set `cursor` to `pointer` on line 71. For normal links it is added by default so the only behaviour we are changing is for internal links by adding pointer cursor -
(Text below uses `div` for internal links and `a` for other/normal links)
https://github.com/Expensify/App/blob/6e8435ec6cba470dae50262439f2bacecea9222f/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js#L68-L71

Need to change line 71 to -
```javascript
style={StyleSheet.flatten([props.style, defaultTextStyle, styles.cursorPointer])}
```


This would solve all the related issues without regression. I tested on local all issues that were related to my issue like https://github.com/Expensify/App/issues/18658, https://github.com/Expensify/App/issues/17488, #16526, are fixed.

Result -

https://github.com/Expensify/App/assets/31413407/808e6e2f-371a-4e68-be19-46b02cf469d0